src/CryptoBank.WebAPI/Features/Auth/Requests/RevokeToken.RequestHandler.cs

```diff
@@ -0,0 +1,55 @@
+using CryptoBank.Common;
+using CryptoBank.Database;
+using CryptoBank.Domain.Models;
+using CryptoBank.WebAPI.Common.Errors.Exceptions;
+using CryptoBank.WebAPI.Features.Auth.Errors;
+using JetBrains.Annotations;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace CryptoBank.WebAPI.Features.Auth.Requests;
+
+public partial class RevokeToken
+{
+    [UsedImplicitly]
+    public class RequestHandler : IRequestHandler<Request>
+    {
+        private readonly IClock _clock;
+        private readonly CryptoBankDbContext _dbContext;
+
+        public RequestHandler(CryptoBankDbContext dbContext, IClock clock)
+        {
+            _dbContext = dbContext;
+            _clock = clock;
+        }
+
+        public async Task Handle(Request request, CancellationToken cancellationToken)
+        {
+            var token = await _dbContext.Tokens.SingleOrDefaultAsync(
+                token => token.RefreshToken == request.RefreshToken, cancellationToken);
+
+            Verify(token);
+
+            await Revoke(token.RefreshToken, cancellationToken);
+        }
+
+        private void Verify([System.Diagnostics.CodeAnalysis.NotNull] Token? token)
+        {
+            if (token == null)
+                throw new LogicConflictException(AuthLogicConflictErrorCode.RefreshTokenDoesNotExist);
```

Comment:
I think `RefreshTokenDoesNotExist` is a logic conflict error because it is a step in the process of verification, and if I validate it in `RequestValidator` there are two operations of getting a token from DB.